_posts/2021-01-31-choosing-randomization-unit.md

```diff
@@ -0,0 +1,353 @@
+
+## User level randomization
+
+Randomizing on the user instead of the session can help us deal with these non-independent sessions. In order to avoid the same pitfalls we faced with session level randomization, we must assume that the users themselves are independent, and do not influence one another's decision to convert <sup><a href="#notes">2</a></sup>. Using our faithful shoppers Terry & Jerry, user level randomization would look something like this:
+
+<p align="center">
+    <img width="100%" src="{{ site.baseurl }}{% link images/choosing-randomization-unit/user_level_split.png %}">
+</p>
+
+We can see that each user is consistently assigned to the same experience across all their sessions. After re-running our simulation with user level randomization, and the same "carryover effects" from before, we recover the results we originally expected:
+
+* The A/B test correctly detects a positive effect 100.0% of the time (at a 95% "confidence" level)
+* The A/B test detects an effect size of 199 bps (~2%) on average
+
+
+# Choosing your metrics
+
+- Now that we've seen the impact of "carryover effects" or nonindependence of sessions, we should talk about metrics
+- In the section above I just looked at session level conversion rate, as this metric is often used due to its simplicity. But where could it lead us astray?
+- What happens if the underlying session conversion rate stayed the same, but session per user increased for treatment?
+    - session conversion rate would actually look smaller
+- Throughout this post, I've assumed that session conversion rate and sessions per user were independent. As discussed in this great post: https://towardsdatascience.com/the-second-ghost-of-experimentation-the-fallacy-of-session-based-metrics-fb65006d30ff, that is usually not the case. Instead, you will typically see something like:
+    - show two curves
+    - now demonstrate higher false positive rate with user level randomization + session level conversion
+    - show how this returns to normal with 
+
+# Wrapping up...
+
+If you think your sessions may not be independent and there could be some "carryover" effect, you should randomize at the user grain instead.
+
+- Example of when this may be the case: product recommendations!
+    - Implications: we underestimate the true effect or even miss detecting it all together
+    - Same can happen with a treatment that has a negative effect, which gets masked by some users returning to the "control" or better experience, but carrying that negative impact with them.
+- Also make sure you look at the distribution of sessions per user. If most users visit close to once, on average, then this will be much less of an issue.
+- Of course, it's difficult to know what the true carryover effect would be, but if you think there is a good chance of some nonindependence existing, and you have the bability to do user level, you should take that option whenever possible to reduce your
+
```

Comment:
re-mention all the other things that should be considered when choosing your randomization grain?